File: python/src/equities_bench/harness/percentiles.py
import math


def sort_in_place(samples: list[int]) -> None:
    samples.sort()


def pct(sorted_samples: list[int], p: float) -> int:
    if not sorted_samples:
        return 0
    i = int(math.floor(p * (len(sorted_samples) - 1)))
    if i < 0:
        i = 0
    if i >= len(sorted_samples):
        i = len(sorted_samples) - 1
    return sorted_samples[i]


def min_v(sorted_samples: list[int]) -> int:
    return sorted_samples[0] if sorted_samples else 0


def max_v(sorted_samples: list[int]) -> int:
    return sorted_samples[-1] if sorted_samples else 0

File: java/src/main/java/com/equities/bench/Percentiles.java
package com.equities.bench;

import java.util.Arrays;

public final class Percentiles {
    private Percentiles() {}

    public static void sortInPlace(long[] samples) {
        Arrays.sort(samples);
    }

    public static long pct(long[] sorted, double p) {
        if (sorted.length == 0) {
            return 0;
        }
        int i = (int) Math.floor(p * (sorted.length - 1));
        if (i < 0) {
            i = 0;
        }
        if (i >= sorted.length) {
            i = sorted.length - 1;
        }
        return sorted[i];
    }

    public static long min(long[] sorted) {
        return sorted.length == 0 ? 0 : sorted[0];
    }

    public static long max(long[] sorted) {
        return sorted.length == 0 ? 0 : sorted[sorted.length - 1];
    }
}
